### PR TITLE
Added UI for adding/deleting user websites

### LINKED ISSUE #1398: Profile: add social, web links
As a user, I'd like to be able to add social links (and other links) to my profile for other users to discover. 

### Invision:
- [rendered profile (desktop)](https://invis.io/9EN908ZQPAX#/314904354_Profile_With_Posts)
- [profile form (dekstop)](https://invis.io/9EN908ZQPAX#/311617870_Edit_Profile)

### Zeplin:
- [rendered profile (desktop)](https://zpl.io/beLnmeN)
- [rendered profile (mobile)](https://zpl.io/bPvWRAy)
- [profile form (mobile)](https://zpl.io/bLxjXYm)

### acceptance criteria
- [ ] display two inputs by default, so users know they can add more than one
- [ ] allow for adding more web sites
- [ ] for recognized social sites, display an icon in the rendered view
  - svg icons are in Zeplin
- [ ] for unrecognized social sites, return a (validation) error (text coming soon)
- [ ] to discourage abuse (link spamming) these links should have `nofollow` attributes

### PR DESCRIPTION
**Pull request by gsidebo**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots 
  - [x] Mobile width screenshots 
  - [x] tag @ferdi or @pdpinch for review 
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1398

#### What's this PR do?
Adds UI for adding/deleting user websites in the profile page

#### How should this be manually tested?
- Try adding websites in the profile edit page then deleting them
- After adding websites, look at the profile page (not in edit mode) and make sure the links all show up

#### Screenshots (if appropriate)
ss 2018-11-29 at 01 43 47
ss 2018-11-28 at 16 20 38
ss 2018-11-28 at 16 20 16

### CODE REVIEW

File: profiles/serializers.py [hunk 1]
```diff
@@ -88,24 +97,77 @@ class Meta:
...
 
-    def validate_site_type(self, value):
+    def validate_url(self, value):
         """
-        Validator for site_type
-
-        NOTE: This is a no-op to allow for the calculation of site_type from the given url.
+        Validator for url. Prepends http protocol to the url if the protocol wasn't already included in the value.
         """
+        if not re.search(r"^http[s]?://", value):
```
Comment (mbertrand) [inline profiles/serializers.py]: This should be case insensitive, if I enter 'HTTP://<something>' I end up with this:
screen shot 2018-11-29 at 1 49 33 pm

**Comment by mbertrand** [comment]
I haven't been able to duplicate the problem with the blank icon again after a few dozen more tries.

**Comment by gsidebo** [comment]
@mbertrand Great catches on all of those! Should be all set now

File: profiles/serializers.py (hunk 1 above)
Comment (mbertrand) [reply profiles/serializers.py]: Still happening, this should fix it:
```python
re.search(r"^http[s]?://", value, re.IGNORECASE)
```

**Comment by pdpinch** [comment]
If the logos came from a creative commons source, we need to attribute them somewhere in the repo. This can be a follow up PR if necessary.

**Comment by pdpinch** [comment]
Nevermind, I see that https://github.com/feathericons/feather/ is licensed under an MIT license. How did you incorporate the feather icons' code?

**Comment by gsidebo** [comment]
New screenshots:

ss 2018-12-03 at 13 29 50
ss 2018-12-03 at 13 29 57
ss 2018-12-03 at 13 30 06

**Comment by gsidebo** [comment]
@pdpinch I added the feather sprite to the project in the images folder and referred to it in the code. That usage is suggested [here](https://github.com/feathericons/feather#svg-sprite) in their docs. I thought it was best to avoid adding a new NPM dependency for our use case (which is pretty limited)

**Comment by pdpinch** [comment]
I think we need to include their license then. I suggest copying the LICENSE file from their repo and dropping it in the directory you created for the social images.

**Comment by gsidebo** [comment]
@pdpinch @Ferdi if we change the placeholder text to indicate the acceptable social sites, mobile users still won't be able to see it (you only get ~20-30 characters). If y'all are OK with that, I am too

ss 2018-12-03 at 16 23 07
ss 2018-12-03 at 16 23 21

**Comment by gsidebo** [comment]
Updated:

ss 2018-12-04 at 10 54 32
ss 2018-12-04 at 10 54 43

**Pull request merged**